cnf-certification-test/certification/suite.go

```diff
@@ -202,3 +207,22 @@ func testContainerCertificationStatusByDigest(env *provider.TestEnvironment, val
 	}
 	testhelper.AddTestResultLog("Non-compliant", failedContainers, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
+
+func testHelmVerion(env *provider.TestEnvironment) {
+	helmchartsReleases := env.HelmChartReleases
+	// Collect all of the failed helm charts
+	failedHelmCharts := [][]string{}
+	var helmChartVersionf float64
+	for _, helm := range helmchartsReleases {
+		helmChartVersion := helm.Chart.Metadata.APIVersion
+		helmChartVersion = strings.Split(helmChartVersion, "+")[0]
+		helmChartVersion = strings.Split(helmChartVersion, "v")[1]
+		helmChartVersionf, _ = strconv.ParseFloat(helmChartVersion, 64)
```

Comment:
there is unreal number this is why im using that parsing

Reply:
Ok, I just suggested that semver package because it would do the parsing for you and the code would look (imho) better:
```
for _, helm := range helmchartsReleases {
  chartName := helm.Chart.Name()
  chartApiVersion := helm.Chart.Metadata.APIVersion

  if !semver.IsValid(chartApiVersion) {
    logrus.Errorf("Failed to parse helm %s version %s, but its major should be v3", chartName, chartApiVersion)
    // Add chart as non-compliant report object...
    ...
    continue
  }

  charApiVersionMajor := semver.Major(chartApiVersion)
  if charApiVersionMajor != "v3" {
    // Add chart as non-compliant report object...
    ...
  } else {
    // Add chart as compliant report object...
    ...
  }
}
```